ddtrace/contrib/internal/google_generativeai/_utils.py

```diff
@@ -28,6 +28,10 @@ def __iter__(self):
         else:
             tag_response(self._dd_span, self.__wrapped__, self._dd_integration, self._model_instance)
         finally:
+            if self._dd_integration.is_pc_sampled_llmobs(self._dd_span):
```

Comment:
This is a bit difficult due to the nature of streamed cases.
In normal non-streamed responses, the user makes the call --> returns a response which contains all data --> we can confidently say the span has finished.
In streamed cases, the main traced function will return a streamed response which does not contain all data and must be consumed before the user (and us) gets all the data we need to collect. So it's not accurate to say the streamed span finishes once the response is returned, but instead when the streamed response has been fully consumed.

Reply:
Thanks for the explanation ! Yes it makes more sense now. 

However I still think we can at least create a wrapper for the finish span method that sets the llmobs tags before finishing the span

llmobs_finish_span(span, input, output, args, kwargs)

And use it instead of span.finish() in the streamed response and in the original. This way we would be setting the LLMObs tags only once in that wrapper.